internal/compiler/expression_tree.rs

```diff
@@ -659,7 +661,13 @@ impl Expression {
                 visitor(&**index);
             }
             Expression::RepeaterIndexReference { .. } => {}
-            Expression::RepeaterModelReference { .. } => {}
+            Expression::RepeaterModelReference { element } => {
+                if let Some(e) = element.upgrade() {
+                    if let Some(r) = &e.borrow().repeated {
+                        visitor(&r.model);
```

Comment:
That doesn't look right.  
1. The expression visitor should only visit its own sub expression.   The model should be visited as its own expression
2. The visit_mut and visit should basically do the same

Reply:
Because the expression is a *reference* to the model.  The model expression itself shouldn't be visited there because it's not to be visited.  You would end up with circular infinite recursion, and the code don't except you to visit expression which are not part of the same expression.
For the same reason that when we have a PropertyReference you don't visit the binding of that property.